pymodbus/datastore/remote.py

```diff
@@ -74,44 +87,20 @@ def __build_mapping(self):
         if self.slave:
             kwargs["slave"] = self.slave
         self.__get_callbacks = {
-            "d": lambda a, c: self._client.read_discrete_inputs(
-                a, c, **kwargs
-            ),
-            "c": lambda a, c: self._client.read_coils(
-                a, c, **kwargs
-            ),
-            "h": lambda a, c: self._client.read_holding_registers(
-                a, c, **kwargs
-            ),
-            "i": lambda a, c: self._client.read_input_registers(
-                a, c, **kwargs
-            ),
+            "d": "read_discrete_inputs",
+            "c": "read_coils",
+            "h": "read_holding_registers",
+            "i": "read_input_registers",
         }
         self.__set_callbacks = {
-            "d5": lambda a, v: self._client.write_coil(
-                a, v, **kwargs
-            ),
-            "d15": lambda a, v: self._client.write_coils(
-                a, v, **kwargs
-            ),
-            "c5": lambda a, v: self._client.write_coil(
-                a, v, **kwargs
-            ),
-            "c15": lambda a, v: self._client.write_coils(
-                a, v, **kwargs
-            ),
-            "h6": lambda a, v: self._client.write_register(
-                a, v, **kwargs
-            ),
-            "h16": lambda a, v: self._client.write_registers(
-                a, v, **kwargs
-            ),
-            "i6": lambda a, v: self._client.write_register(
-                a, v, **kwargs
-            ),
-            "i16": lambda a, v: self._client.write_registers(
-                a, v, **kwargs
-            ),
+            "d5": "write_coil",
```

Comment:
This is not the same...but might be the reason I saw a getattr earlier.

We do not want that, apart from that what does it have to do with getValues being async ?

Reply:
this is mainly due lamdas and await don't work together. It can be redone without getattr use most likely easily.